src/main/java/duke/BH.java

```diff
@@ -0,0 +1,168 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represent a chatbot called BH
+ * It is able to read input and proceed with different kind of task and save all the tasks
+ */
+public class BH {
+    private ArrayList<Task> list;
+    private Storage storage;
+    private Scanner sc = new Scanner(System.in);
+    private Ui ui;
+    private static String filePath = "/Users/brandonrhan/Downloads/NUS/CS2103/ip/data/duke.txt";
+
+    /**
+     * constructor of BH
+     * @throws DukeException if the filepath is invalid
+     */
+    public BH() throws DukeException {
+        this.storage = new Storage(filePath);
+        this.list = this.storage.load();
+        this.ui = new Ui();
+    }
+
+    /**
+     * continue reading input until a bye is detected
+     * if input starts with list, print out all tasks in the list
+     * if input starts with mark, mark the task as done
+     * if input starts with unmark, unmark the task as not done
+     * if input starts with todo, deadline or event, create a corresponding task and add to list
+     * if input starts with delete, delete the corresponding task
+     * if input starts with check, check all the task on the same date
+     * @throws DukeException if wrong input is detected
```

Comment:
I think there should be an empty line between the description and the parameter section!